packages/mobile/src/utils/keyStore.ts

```diff
@@ -1,13 +1,34 @@
 import RNSecureKeyStore, { ACCESSIBLE } from 'react-native-secure-key-store'
+import { Sentry } from 'react-native-sentry'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'utils/keystore'
 
 export async function setKey(key: string, value: string) {
-  return RNSecureKeyStore.set(key, value, { accessible: ACCESSIBLE.WHEN_UNLOCKED })
+  try {
+    Logger.debug(TAG, `Setting key ${key} in keystore`)
+    await RNSecureKeyStore.set(key, value, {
+      accessible: ACCESSIBLE.AFTER_FIRST_UNLOCK_THIS_DEVICE_ONLY,
```

Comment:
Why changing the `accessible` option now?

Reply:
I was just experimenting before I realized this only affects iOS. This option seems right to me, WDYT?

Reply:
This has a subtle implication when restoring the phone from a backup. 

If using one of the options with the `_THIS_DEVICE_ONLY` suffix, the keychain item won't be backed up. Which means users will lose their mnemonic if they restore their phone (though I know they should have manually backed up their key). Though the `AsyncStorage` content will be restored.

This doesn't sound right to me as is and we should make a careful consideration about this (and same for the geth data folder and `AsyncStorage` content).

More about this:
https://developer.apple.com/documentation/security/ksecattraccessibleafterfirstunlockthisdeviceonly

Reply:
Yep that was intentional, I felt that the backup key should not be synced to Apple's servers.

Reply:
Will circle back on this directly on Slack.